packages/@webex/plugin-meetings/src/peer-connection-manager/util.js

```diff
@@ -16,4 +42,71 @@ PeerConnectionUtils.convertCLineToIpv4 = (sdp) => {
 
   return replaceSdp;
 };
+
+/**
+ * estimate profile levels for max-fs & max-mbps values
+ * @param {string} sdp
+ * @param {number} maxFsValue
+ * @returns {string}
+ */
+PeerConnectionUtils.estimateProfileLevels = (sdp, maxFsValue) => {
+  // converting with ts-sdp parser, no munging
+  const parsedSdp = parse(sdp);
+
+  parsedSdp.avMedia.forEach((media) => {
+    if (media.type === 'video') {
+      media.codecs.forEach((codec) => {
+        if (codec.name?.toUpperCase() === 'H264') {
+          // there should really be just 1 fmtp line, but just in case, we process all of them
+          codec.fmtParams = codec.fmtParams.map((fmtp) => {
+            const parsedRegex = fmtp.match(/(.*)profile-level-id=(\w{4})(\w{2})(.*)/);
+
+            if (parsedRegex && parsedRegex.length === 5) {
+              const stuffBeforeProfileLevelId = parsedRegex[1];
+              const profile = parsedRegex[2].toLowerCase();
+              const levelId = parseInt(parsedRegex[3], 16);
+              const stuffAfterProfileLevelId = parsedRegex[4];
+
+              if (!maxFsForProfileLevel[levelId]) {
+                throw new Error(
+                  `found unsupported h264 profile level id value in the SDP: ${levelId}`
+                );
+              }
+
+              if (maxFsForProfileLevel[levelId] === maxFsValue) {
+                // profile level already matches our desired max-fs value, so we don't need to do anything
+                return fmtp;
+              }
+              if (maxFsForProfileLevel[levelId] < maxFsValue) {
+                // profile level has too low max-fs, so we need to override it (this is upgrading)
+                return `${fmtp};max-fs=${maxFsValue};max-mbps=${maxFsValue * framePerSecond}`;
+              }
+
+              // profile level has too high max-fs value, so we need to use a lower level
+
+              // find highest level that has the matching maxFs
+              const newLevelId = Object.keys(maxFsForProfileLevel)
+                .reverse()
```

Comment:
I don't think reversing this makes any difference?

Reply:
We are reversing because we wanted to use the lower level value, For an example For some profile levels, we have similar max-fs, but we wanted to use the one where max-mbps value is higher.
@marcin-bazyl correct me if I am wrong ?

Reply:
We are reversing because we wanted to use the lower level value, For an example For some profile levels, we have similar max-fs, but we wanted to use the one where max-mbps value is higher.
@marcin-bazyl , correct me if I am wrong ?